### ISSUE TITLE
enhancement: mount cgroups filesystem in exec and java driver

### ISSUE THREAD

**Issue by ferusm** [hashicorp/nomad#5376]
### Nomad version
Nomad v0.8.7 (21a2d93eecf018ad2209a5eab6aae6c359267933+CHANGES)

### Operating system and Environment details
OS: Debian 9.7
Java: OpenJDK 11

### Issue
When starting job with java driver, 'resources' stanza has no effect.
Jvm container debug output: "Required cgroup memory subsystem not found"

### Reproduction steps
Run job

### Job file

```
job "..." {
 datacenters = ["..."]
 type = "service"
  group "..." {
    count = 3
    task "service" {
      driver = "java"
      config {
        jar_path = "/opt/.../....jar"
        jvm_options = ["-Xlog:os+container=trace"]
      }
      resources {
        cpu = 300
        memory = 1000
      }
    }
  }
}
```

**Comment by cgbaker** [comment]
The `resources` stanza does have an effect, it's just that the Java runtime isn't aware of it. We don't mount the necessary cgroups for the new Java container functionality to work. I'll file an issue on our roadmap to look into this, and I'll leave this ticket open for updates.

Thanks for bringing this to our attention.

**Comment by rkettelerij** [comment]
>  We don't mount the necessary cgroups for the new Java container functionality to work.

Really? So `-XX:+UseCGroupMemoryLimitForHeap` (Java version < 8u191) and/or `-XX:+UseContainerSupport` (Java version > 8u191, and Java 10, 11) doesn't have any effect?

**Comment by ferusm** [comment]
Seems to be yes. There are no corresponding files in the "container" in / system / fs / group. But on the host, the corresponding group hierarchy is created.

**Comment by cgbaker** [comment]
That's correct. That's the warning message: 
```
Required cgroup memory subsystem not found
```
from approximately here: http://hg.openjdk.java.net/jdk/jdk/file/03f2bfdcb636/src/hotspot/os/linux/osContainer_linux.cpp#l330

**Comment by rkettelerij** [comment]
>  I'll file an issue on our roadmap to look into this

@cgbaker is this currently on the roadmap? See also https://github.com/hashicorp/nomad/issues/5940

**Comment by Oloremo** [comment]
Any updates?

**Comment by tgross** [comment]
This was shipped in the big driver refactor in 0.9.0, where now `java` shares an executor with `exec`.
